src/mame/drivers/cinemat.cpp

```diff
@@ -371,25 +371,20 @@ static INPUT_PORTS_START( spacewar )
 	PORT_BIT( 0x08, IP_ACTIVE_LOW, IPT_BUTTON3 ) PORT_PLAYER(2)
 	PORT_BIT( 0x10, IP_ACTIVE_LOW, IPT_BUTTON1 ) PORT_PLAYER(1)
 	PORT_BIT( 0x20, IP_ACTIVE_LOW, IPT_BUTTON3 ) PORT_PLAYER(1)
-	PORT_DIPNAME( 0x40, 0x40, DEF_STR( Unknown ) )
-	PORT_DIPSETTING(    0x40, DEF_STR( Off ) )
-	PORT_DIPSETTING(    0x00, DEF_STR( On ) )
-	PORT_BIT( 0x80, IP_ACTIVE_LOW, IPT_COIN1 ) PORT_CHANGED_MEMBER(DEVICE_SELF, cinemat_state,coin_inserted, 0)
+	PORT_BIT( 0x40, IP_ACTIVE_LOW, IPT_OTHER ) PORT_NAME("Reset Playfield") PORT_CODE(KEYCODE_R)
```

Comment:
You should really avoid `IPT_OTHER` with overridden assignments in the driver.  Remember people use MAME with input devices other than keyboards, and people do use general input assignments.

Is this a player input or an operator input?  If it’s a player input, it should probably be a button.  If it’s an operator input, use one of the service buttons.

Reply:
That's a bit of a weird arrangement. The trouble with buttons is they'll get assigned to one player or the other. There's a case to be made for using `IPT_KEYPAD` for that, but it still has the disadvantage of needing the default assignment set in the driver. It would probably be helped by a clickable layout, too.